### DISCUSSION TITLE
Is there an upgrade guide from v4 to v5?

### DISCUSSION THREAD

**Discussion by dmerrick** [argoproj-labs/hera#562] [Ideas]
I'm excited about v5 and have started updating my codebase for it. It's a long process though since I have yet to find a concise guide about how to move my resources over.

Some things I've learned:
* most classes have been moved to `hera.workflows`
* there are some exceptions, like `RetryStrategy` lives in `hera.workflows.retry_strategy`
* internal enums have been renamed, so `RetryPolicy.OnError` becomes `RetryPolicy.on_error`
* `WorkflowService` is now `WorkflowsService` (this was a hard one to spot!)
* `Task` now takes only one positional argument

I'm sure there are a ton more, because I'm not done with my code and I don't even touch a lot of the features available

**Comment by flaviuvadan** [comment]
Thanks for the great suggestion @dmerrick! I can put together a quick guide in the next few days.
